hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/record/AbstractNearCacheRecord.java

```diff
@@ -42,6 +42,7 @@
             AtomicLongFieldUpdater.newUpdater(AbstractNearCacheRecord.class, "recordState");
 
     protected long creationTime = TIME_NOT_SET;
+    protected int partitionId;
```

Comment:
Yes, that 's my understanding. Otherwise there is no guarantee that the values set `onRecordCreate` on one thread will be visible to other threads. Would it be possible to construct the `Record` in a single step and turn the fields `final`?